elixir/config/runtime.exs

```diff
@@ -1,22 +1,31 @@
 import Config
 
 if config_env() == :prod do
-  import Domain.Config, only: [compile_config!: 1]
+  import Domain.Config, only: [compile_config!: 1, compile_config: 1]
 
   ###############################
   ##### Domain ##################
   ###############################
 
-  config :domain, Domain.Repo,
-    database: compile_config!(:database_name),
-    username: compile_config!(:database_user),
-    hostname: compile_config!(:database_host),
-    port: compile_config!(:database_port),
-    password: compile_config!(:database_password),
-    pool_size: compile_config!(:database_pool_size),
-    ssl: compile_config!(:database_ssl_enabled),
-    ssl_opts: compile_config!(:database_ssl_opts),
-    parameters: compile_config!(:database_parameters)
+  config :domain,
+         Domain.Repo,
+         [
+           {:database, compile_config!(:database_name)},
+           {:username, compile_config!(:database_user)},
+           {:port, compile_config!(:database_port)},
+           {:pool_size, compile_config!(:database_pool_size)},
+           {:ssl, compile_config!(:database_ssl_enabled)},
+           {:ssl_opts, compile_config!(:database_ssl_opts)},
+           {:parameters, compile_config!(:database_parameters)}
+         ] ++
+           if(compile_config(:database_password),
+             do: [{:password, compile_config!(:database_password)}],
+             else: []
+           ) ++
+           if(compile_config(:database_socket_dir),
+             do: [{:socket_dir, compile_config!(:database_socket_dir)}],
+             else: [{:hostname, compile_config!(:database_host)}]
+           )
```

Comment:
Just wondering if this complexity is needed?  I've not ever used `peer` auth with Postgresql (at least that I remember), so I may be missing something, but in a quick search it seems that if you provide a password when using the `peer` auth method, it will be ignored and if the comment in `definitions.ex` file is correct, then providing a `socket_dir` should take precedence over providing a `host` right?

Reply:
cc original author, @oddlama

Reply:
If we move this to the common block, `compile_config!` will enforce the user to provide the `DATABASE_PASSWORD` environment variable at all times, even if the password is not actually required. I found it counter intuitive to enforce this, as it also communicates to the user that this is a necessary variable - which it isn't. IMO this would just move the complexity from here to somewhere else (the user)

Reply:
I personally just want peer authentication to work at all, so if you'd rather like to remove the conditional then that's fine, too. Although I don't think I am allowed to update this branch.